### DISCUSSION TITLE
How do i disable SSR for a particular route?

### DISCUSSION THREAD

**Discussion by xiongsongsong** [nuxt/nuxt#17465] [Questions]
### Environment

latest

### Describe the bug

ssr:false not work.

### Reproduction

# nuxt.config.ts

```typescript
import { defineNuxtConfig } from 'nuxt3'

export default defineNuxtConfig({
// not work
  routers: {
    '/news': {
       ssr: true
    },
    '/admin': {
      ssr: false, 
    }
  }
})
```

**Comment by danielroe** [comment]
We're going to be implementing this on a per-route basis - see nuxt/nuxt#16078.

A current workaround is to set `req.spa = true` in a `server/middleware/test.js` file.

```js
export default (req) => {
  if (req.url === '/admin') {
    req.spa = true
  }
}
```

**Reply by xiongsongsong** [reply]
thank's!

**Reply by andrewspy** [reply]
Just want to point out that you need to `export default async`, else it won't work, i.e.,

```js
export default async (req) => {
  if (req.url === '/admin') {
    req.spa = true
  }
}
```

**Reply by andrewspy** [reply]
@danielroe It seems like `req.spa` is no longer working in nuxt3-rc.3, may I know what's the way forward?

**Reply by andrewspy** [reply]
@danielroe I have tried to set the experimental `routes` with the following in nuxt3-rc.3, and it's still not working. Did I miss out anything?

```js
{
  routes: {
    '/spa-only/**': { ssr: false }
  }
}
```

**Reply by derz** [reply]
@andrewspy 

```js
export default defineEventHandler((event) => {
	if (...) {
		event.req.headers["x-nuxt-no-ssr"] = "true";
	}
});
```
should work

**Reply by azohc** [reply]
`ssr: false` is working for me, but its giving me a TypeScript error. What's up here? Is it Nitro that's defining the options for `routeRules`? The TS error is "Object literal may only specify known properties, and 'ssr' does not exist in type ..."

For now I'm just using a @ts-expect-error comment above it and all is fine
